pkg/fleet/installer/service/datadog_agent_windows.go

```diff
@@ -8,22 +8,99 @@
 // Package service provides a way to interact with os services
 package service
 
-import "context"
+import (
+	"context"
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/fleet/internal/winregistry"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
 
-// SetupAgent noop
-func SetupAgent(_ context.Context) error {
-	return nil
+func msiexec(target, operation string, args []string) (err error) {
+	programData, err := winregistry.GetProgramDataDirForProduct("Datadog Installer")
+	if err != nil {
+		return nil
+	}
+	updaterPath := filepath.Join(programData, "datadog-agent", target)
+	msis, err := filepath.Glob(filepath.Join(updaterPath, "datadog-agent-*-1-x86_64.msi"))
+	if err != nil {
+		return nil
+	}
+	if len(msis) > 1 {
+		return fmt.Errorf("too many MSIs in package")
+	} else if len(msis) == 0 {
+		return fmt.Errorf("no MSIs in package")
+	}
+
+	tmpDir, err := os.MkdirTemp(os.TempDir(), fmt.Sprintf("install-%s-*", filepath.Base(msis[0])))
```

Comment:
On linux we had issues with `noexec` partition when we did things like this. Any risk of similar issues on windows where tmp is special and you can't exec from it ?